### ISSUE TITLE
Remove method ufc::finite_element::map_dofs

### ISSUE THREAD

**Issue by garth-wells** [FEniCS/ffcx#4]
This function (it seems) interpolates f(x) into an FE space. It is often used magically, causing some user confusion.

It would be simpler and cleaner to just use point-wise eval of functions.

**Comment by garth-wells** [comment]
Has this been addressed, at least for now? @chrisrichardson @blechta

**Comment by blechta** [comment]
`map_dofs` has been renamed to:
```
    /// Map values of field from physical to reference space which has
    /// been evaluated at points given by tabulate_reference_dof_coordinates.
    void (*transform_values)(ufc_scalar_t* restrict reference_values,
                             const ufc_scalar_t* restrict physical_values,
                             const double* restrict coordinate_dofs,
                             int cell_orientation,
                             const ufc_coordinate_mapping* cm);
```

**Comment by blechta** [comment]
I don't think that it's appropriate to remove it.

**Comment by garth-wells** [comment]
I'm closing this issue because the function has been renamed, and the unresolved technical issues are addressed in #33.
